### PR TITLE
Fix SyncResolveError values

### LINKED ISSUE #676: syncErrorHandler - provide better details in SyncError
I'm using the `flutter_flexible_sync` example as the basis of a POC I'm working on, and after adding sync capabilities with Atlas, I'm not able to make the app work in offline mode.

Am I doing something wrong?

This is the code I'm using to create the `Realm` object.
``` dart
Future<Realm> createRealm(String appId) async {
  final user = await App(AppConfiguration(appId)).logIn(
    Credentials.anonymous(),
  );

  final realm = Realm(
    Configuration.flexibleSync(
      user,
      [
        // ...
      ],
    ),
  );
  print("Created local realm db at: ${realm.config.path}");

  realm.subscriptions.update((mutableSubscriptions) {
    mutableSubscriptions.add(realm.all<Car>());
    mutableSubscriptions.add(realm.all<Person>());
  });
  await realm.subscriptions.waitForSynchronization();
  print("Syncronization completed for realm: ${realm.config.path}");

  return realm;
}
```

And this is the issue generated when trying to launch the app in offline mode:

```
flutter: [INFO] Realm: Connection[1]: Session[1]: client_reset_config = false, Realm exists = true, client reset = false
flutter: [ERROR] Realm: Failed to resolve 'ws.realm.mongodb.com:443': Host not found (authoritative)
```

### PR DESCRIPTION
**Pull request by desistefanova**
`SyncResolveError` enum improvement. 
Added error code numbers according to the codes in C-API - https://github.com/realm/realm-core/pull/6186

### CODE REVIEW

**Review by blagoev — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 **This project is in Release Candidate stage.**
 
 ### Enhancements
-* None
+* Added values to items of `SyncResolveErrorCode` ([#1131](https://github.com/realm/realm-dart/pull/1131).
```
Comment (blagoev) [inline CHANGELOG.md:6]: this could be reworded cause now it is not clear what the improvement is.

**Review by blagoev — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (desistefanova) [reply CHANGELOG.md]: Is this better? "Added specific codes to `SyncResolveErrorCode` enum`s items."

Comment (blagoev) [reply CHANGELOG.md]: We should clarify to a developer what the enhancement is.

**Review by blagoev — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (desistefanova) [reply CHANGELOG.md]: Moved to Internal section.

**Review by blagoev — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 ### Enhancements
 * Add `App.reconnect()` providing a hint to Realm to reconnect all sync sessions.
+* Add `Realm.refresh()` and `Realm.refreshAsync()` support. ([#1046](https://github.com/realm/realm-dart/pull/1046))
```
Comment (nielsenko) [inline CHANGELOG.md:7]: 👍

File: CHANGELOG.md [hunk 3]
```diff
@@ -54,7 +57,6 @@
 ### Enhancements
 * Support setting `maxNumberOfActiveVersions` when creating a `Configuration`. ([#1036](https://github.com/realm/realm-dart/pull/1036))
 * Add List.move extension method that moves an element from one index to another. Delegates to ManagedRealmList.move for managed lists. This allows notifications to correctly report moves, as opposed to reporting moves as deletes + inserts. ([#1037](https://github.com/realm/realm-dart/issues/1037))
-* Add `Realm.refresh()` and `Realm.refreshAsync()` support. ([#1046](https://github.com/realm/realm-dart/pull/1046))
```
Comment (nielsenko) [inline CHANGELOG.md:57]: 👍

File: lib/src/session.dart [hunk 4]
```diff
@@ -551,23 +551,32 @@ enum SyncSessionErrorCode {
 /// These errors will be reported via the error handlers of the affected sessions.
 enum SyncResolveErrorCode {
   /// Host not found (authoritative).
-  hostNotFound,
+  hostNotFound(1),
```
Comment (nielsenko) [inline lib/src/session.dart:554]: Previously you counted from 0. Was that a bug? Or is this?

**Review by nielsenko — COMMENTED** [review]
Got a question about the actual int codes, but otherwise LGTM

File: lib/src/session.dart (hunk 4 above)
Comment (blagoev) [reply lib/src/session.dart:554]: previously it was a bug

Comment (blagoev) [reply lib/src/session.dart:554]: see `realm_sync_error_resolve` in realm.h

**Review by nielsenko — APPROVED** [review]

**Pull request merged**